### PR TITLE
Ansible role that sets up a mariadb server instance

### PR DESCRIPTION
**Pull request by Scriptkiddi**
Added defaults that are equal to the default values from my-huge.cnf
besides one value that "skip-networking" is by default true so that
every network connection will be rejected when the server is run with
default values

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Role Name
```
Comment (SlothOfAnarchy) [inline README.md]: set role name

File: meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+galaxy_info:
+  author: Fritz Otlinghaus
+  description: Ansible role to configure and start an mariadb server
```
Comment (SlothOfAnarchy) [inline meta/main.yml]: "a" instead of "an"

File: templates/my.cnf.j2 [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+# Default options are read from the following files in the given order:
+# More information at: http://dev.mysql.com/doc/mysql/en/option-files.html
+#
+# In this file, you can use all long options that a program supports.
+# If you want to know which options a program supports, run the program
+# with the "--help" option.
+
+# The following options will be passed to all MySQL clients
+[client]
+#password   = your_password
```
Comment (SlothOfAnarchy) [inline templates/my.cnf.j2]: add password var, maybe even mark it mandatory

File: templates/my.cnf.j2 [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+read_buffer_size = {{- mariadb_mysqld_read_buffer_size -}} 
+read_rnd_buffer_size = {{- mariadb_mysqld_read_rnd_buffer_size -}}
+myisam_sort_buffer_size = {{- mariadb_mysqld_myisam_sort_buffer_size -}}
+thread_cache_size = {{- mariadb_mysqld_thread_cache_size -}}
+query_cache_size = {{- mariadb_mysqld_query_cache_size -}} 
+# Try number of CPU's*2 for thread_concurrency
+thread_concurrency = {{- mariadb_mysqld_thread_concurrency -}}
+
+# Point the following paths to a dedicated disk
+#tmpdir     = /tmp/
```
Comment (SlothOfAnarchy) [inline templates/my.cnf.j2]: add tempdir var

File: templates/my.cnf.j2 [hunk 5]
```diff
@@ -0,0 +1,163 @@
+# Example MariaDB config file for very large systems.
```
Comment (SlothOfAnarchy) [inline templates/my.cnf.j2:2]: mark as ansible_managed

File: templates/my.cnf.j2 [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+# binary logging - not required for slaves, but recommended
+#log-bin=mysql-bin
+#
+# binary logging format - mixed recommended 
+#binlog_format=mixed
+
+# Uncomment the following if you are using InnoDB tables
+#innodb_data_home_dir = @localstatedir@
+#innodb_data_file_path = ibdata1:2000M;ibdata2:10M:autoextend
+#innodb_log_group_home_dir = @localstatedir@
```
Comment (SlothOfAnarchy) [inline templates/my.cnf.j2]: add innodb support

File: tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+---
+# tasks file for .
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: remove or correct comment

File: README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+
+## Requirements
+
+An apt based packagmanager and systemd
+
+
+## Role Variables
+
+Read the mariadb doc to get to know what vars can be set
```
Comment (SlothOfAnarchy) [inline README.md]: link?

File: templates/my.cnf.j2 [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+# (and different from the master)
+# defaults to 2 if master-host is set
+# but will not function as a slave if omitted
+#server-id       = 2
+#
+# The replication master for this slave - required
+#master-host     =   <hostname>
+#
+# The username the slave will use for authentication when connecting
+# to the master - required
```
Comment (SlothOfAnarchy) [inline templates/my.cnf.j2]: you **could** add vars for replication

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Scriptkiddi) [reply README.md]: done

**Review by SlothOfAnarchy — APPROVED** [review]

File: templates/my.cnf.j2 [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+# Replication Master Server (default)
+# binary logging is required for replication
+log-bin={{ mariadb_mysqld_log_bin }}
+
+# required unique id between 1 and 2^32 - 1
+# defaults to 1 if master-host is not set
+# but will not function as a master if omitted
+server-id   = {{ mariadb_mysqld_server_id }}
+
+# Replication Slave (comment out master section to use this)
```
Comment (dasJ) [inline templates/my.cnf.j2:67]: @Scriptkiddi Why the fuck is there a huge commented out area?

**Pull request merged**